server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -320,6 +326,19 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
         // catch early terminated exception and rethrow?
         Bits liveDocs = ctx.reader().getLiveDocs();
         BitSet liveDocsBitSet = getSparseBitSetOrNull(liveDocs);
+        if (isApproximateableRangeQuery()) {
+            ApproximateableQuery query = ((ApproximateableQuery) ((IndexOrDocValuesQuery) searchContext.query()).getIndexQuery());
+            if (searchContext.size() > 10_000) ((ApproximatePointRangeQuery) query.getApproximationQuery()).setSize(searchContext.size());
```

Comment:
nitpicks, can we please extract to a constant here & break lines?

Reply:
Specifically, I think it shouldn't be a constant at all. The value of `10_000` is just the default for `track_total_hits`, but can be specified in the request.

Rather, it should probably be something like:

```
(ApproximatePointRangeQuery) query.getApproximationQuery()).setSize(Math.max(searchContext.size(), searchContext.trackTotalHitsUpTo());
```